src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -167,26 +167,26 @@ private function registerCommonConfiguration(ContainerBuilder $container, array
         $container->setParameter('api_platform.eager_loading.fetch_partial', $config['eager_loading']['fetch_partial']);
         $container->setParameter('api_platform.eager_loading.force_eager', $config['eager_loading']['force_eager']);
         $container->setParameter('api_platform.collection.exists_parameter_name', $config['collection']['exists_parameter_name']);
-        $container->setParameter('api_platform.collection.order', $config['collection']['order']);
+        $container->setParameter('api_platform.collection.order', $config['defaults']['order'] ?? $config['collection']['order']);
```

Comment:
This change was wrong! `defaults.order` is supposed to be the global resource default for https://api-platform.com/docs/core/default-order/#overriding-default-order i.e. an array like `{"foo":"ASC"}` or `["foo", "bar"]`, not to be confused with `collection.order` that is a string `"ASC"` or `"DESC"`!

I first wrote this:

```yaml
api_platform:
    collection:
        order: DESC
    defaults:
        order: [createdAt, id]
```

but got this error:

> Argument 1 passed to ApiPlatform\\Core\\Bridge\\Doctrine\\Orm\\Extension\\OrderExtension::__construct() must be of the type string or null, array given

Fortunately I could work around by using the longer:

```yaml
api_platform:
    collection:
        order: DESC
    defaults:
        attributes:
            order: [createdAt, id]
```

but still...

Can this change be reverted as a bug fix or would it be a BC break now?

Reply:
IMO it's more a bugfix since this change is wrong.

Reply:
Thanks, I opened #4170